crt_portal/cts_forms/views.py

```diff
@@ -524,46 +525,70 @@ def get(self, request):
             'all_ids_count': all_ids_count,
-            'assign_form': assign_form,
+            'bulk_actions_form': bulk_actions_form,
         }
         return render(request, 'forms/complaint_view/actions/index.html', output)
 
     def post(self, request):
-        assign_form = BulkAssign(request.POST)
+        bulk_actions_form = BulkActions(request.POST)
         return_url_args = request.POST.get('next', '')
         selected_all = request.POST.get('all', '') == 'all'
         confirm_all = request.POST.get('confirm_all', '') == 'confirm_all'
         ids = request.POST.get('ids', '').split(',')
 
-        if assign_form.is_valid():
-            assignee = assign_form.cleaned_data['assigned_to']
+        if bulk_actions_form.is_valid():
             if confirm_all:
                 requested_query = self.reconstruct_query(return_url_args)
             else:
                 requested_query = Report.objects.filter(pk__in=ids)
 
-            # update activity log _before_ we update the assignee so
-            # that we have access to the original assignee
-            for report in requested_query:
-                original = report.assigned_to or "None"
-                description = f'Updated from "{original}" to "{assignee}"'
-                add_activity(request.user, "Assigned to:", description, report)
-
-            number = requested_query.update(assigned_to=assignee)
+            updated_data = bulk_actions_form.get_updates()
+            comment_string = updated_data.pop('comment', None)
+            summary_string = updated_data.pop('summary', None)
 
-            description = f"{number} records have been assigned to {assignee}"
-            messages.add_message(request, messages.SUCCESS, description)
+            # update activity log _before_ we update fields so
+            # that we still have access to the original field
+            for report in requested_query:
+                bulk_actions_form.update_activity_stream(request.user, report)
+
+            if comment_string:
+                kwargs = {
+                    'is_summary': False,
+                    'note': comment_string,
+                    'author': request.user.username,
+                }
+                comment = CommentAndSummary.objects.create(**kwargs)
+                for report in requested_query:
+                    report.internal_comments.add(comment)
+                    add_activity(request.user, 'Added comment: ', comment_string, report)
+
+            if summary_string:
+                kwargs = {
+                    'is_summary': True,
+                    'note': summary_string,
+                    'author': request.user.username,
+                }
+                comment = CommentAndSummary.objects.create(**kwargs)
+                for report in requested_query:
+                    report.internal_comments.add(comment)
+                    add_activity(request.user, 'Added summary: ', summary_string, report)
```

Comment:
I debated on whether or not to override the original summary if extant but I think here it's best added as a entirely new summary, so that the last summary is still available in the database if needed. Note that `Report.get_summary` gets the last modified summary anyway so things should just continue to work. @Jkrzy curious as to your thoughts.

Reply:
I think we may have to revise this -- _editing_ a summary for a bulk updated report will change the summary for all the other (bulk updated) reports.

Two options:
- Create a new summary for each report instead (will not overwrite previous summary)
- Get or create summary for each report (this will overwrite previous summary)

Reply:
Good catch -- each `Report` needs it's own `Summary` instance and this should be modeled as a `ForeignKeyField` instead of `ManyToManyField`.

Option 2 you proposed is how we handle updating individual `Reports`.
We should use the same process here,  found in `ReportEditForm.save` using `CommentAndSummary.objects.update_or_create`

Reply:
Sure thing! Thanks -- I copied the process there for consistency.